src/game2048/board.rs

```diff
@@ -0,0 +1,317 @@
+use crate::world::Direction;
+use alloc::vec::Vec;
+use rand::prelude::*;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct World {}
+
+#[derive(Debug)]
+pub struct Tile {
+    pub val: Option<u64>,
+    pub changed: bool,
+    pub row: usize,
+    pub col: usize,
+}
+
+impl Tile {
+    pub fn new(row: usize, col: usize) -> Self {
+        Self {
+            val: None,
+            changed: false,
+            row,
+            col,
+        }
+    }
+
+    fn can_merge(&self, other: &Self) -> bool {
+        if let (Some(ref v1), Some(ref v2)) = (self.val, other.val) {
+            if v1 == v2 {
+                return true;
+            }
+        }
+        false
+    }
+}
+
+pub(crate) struct Board {
+    rows: usize,
+    cols: usize,
+    tiles: Vec<Tile>,
+    rng: rand::rngs::SmallRng,
+}
+
+impl Board {
+    pub fn new(rows: usize, cols: usize) -> Self {
+        let mut tiles = Vec::with_capacity(rows * cols);
+
+        for row in 0..rows {
+            for col in 0..cols {
+                tiles.push(Tile::new(row, col));
+            }
+        }
+
+        Self {
+            rows,
+            cols,
+            tiles,
+            rng: rand::rngs::SmallRng::from_seed([0; 32]),
```

Comment:
I know its the same in snake, but it would be really nice to have some sort of hardware random. Perhaps current cpu clockspeed or so. Perhap you have an idea?

Reply:
getrandom can use `rdrand` instruction to read random value on x86 cpu, then we can obtain a SmallRng by calling `SmallRng::from_entropy`

https://docs.rs/getrandom/0.2.3/getrandom/#rdrand-on-x86

Reply:
now get a random seed by calling `core::arch::x86_64::_rdtsc()`